compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -1175,9 +1179,8 @@ object Semantic:
         // Note that a parameterized trait may only get parameters from the class that extends the trait.
         // A trait may not supply constructor arguments to another trait.
         if !klass.is(Flags.Trait) then
-          for parent <- klass.parentSyms if parent.hasSource do doPromote(parent.asClass, klass, isHotSegment)
```

Comment:
Why is it OK to remove this? We don't need to promote super-trait segments? Is it because it is duplicated by the promotion of mixins below?

Reply:
Yes, with the changes we avoid some duplicate check (with cache they are not expensive though):

``` Scala
trait A

class B extends A

class C extends B with A
```